File: app/src/main/java/hoo/etahk/common/view/AlwaysAutoCompleteTextView.java
package hoo.etahk.common.view;

import android.content.Context;
import android.graphics.Rect;
import android.util.AttributeSet;
import android.support.v7.widget.AppCompatAutoCompleteTextView;


/**
 * Source: https://stackoverflow.com/questions/2126717/android-autocompletetextview-show-suggestions-when-no-text-entered
 * TODO("Convert to Kotlin")
 */
public class AlwaysAutoCompleteTextView extends AppCompatAutoCompleteTextView {

    private boolean isInit = true;

    public AlwaysAutoCompleteTextView(Context context) {
        super(context);
    }

    public AlwaysAutoCompleteTextView(Context arg0, AttributeSet arg1) {
        super(arg0, arg1);
    }

    public AlwaysAutoCompleteTextView(Context arg0, AttributeSet arg1, int arg2) {
        super(arg0, arg1, arg2);
    }

    @Override
    public boolean enoughToFilter() {
        return true;
    }

    @Override
    protected void onFocusChanged(boolean focused, int direction,
                                  Rect previouslyFocusedRect) {
        super.onFocusChanged(focused, direction, previouslyFocusedRect);
        if (!isInit) {
            if (focused && getAdapter() != null) {
                if (!isInit) {
                    performFiltering(getText(), 0);
                }
            }
        } else {
            isInit = false;
        }
    }
}

File: app/src/main/java/hoo/etahk/common/view/AlwaysAutoCompleteTextView.kt
package hoo.etahk.common.view

import android.content.Context
import android.graphics.Rect
import androidx.appcompat.widget.AppCompatAutoCompleteTextView
import android.util.AttributeSet


/**
 * Source: https://stackoverflow.com/questions/2126717/android-autocompletetextview-show-suggestions-when-no-text-entered
 */
class AlwaysAutoCompleteTextView : AppCompatAutoCompleteTextView {

    private var isInit = true

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun enoughToFilter(): Boolean = true

    override fun onFocusChanged(focused: Boolean, direction: Int, previouslyFocusedRect: Rect?) {
        super.onFocusChanged(focused, direction, previouslyFocusedRect)
        if (!isInit) {
            if (focused && adapter != null) {
                if (!isInit) {
                    performFiltering(text, 0)
                }
            }
        } else {
            isInit = false
        }
    }
}
